stdlib/public/runtime/Float16Support.cpp

```diff
@@ -150,6 +151,29 @@ SWIFT_RUNTIME_EXPORT unsigned short __truncdfhf2(double d) {
   return __gnu_f2h_ieee(f);
 }
 
+// Convert from Float16 to long double.
+//
+// Since Float32 covers the entire range
+// of Float16 values and since we already know how to convert Float32 to long
+// double (which, at least on x86, doesn't involve function calls), we just
+// let the compiler do the latter part for us.
+//
+// There's no risk of rounding problems from the double conversion, because
+// we're extending.
+SWIFT_RUNTIME_EXPORT long double __extendhfxf2(short h) {
```

Comment:
If I'm reading the conditionals correctly, this entire file is compiled only for
* Android on ARM
* Windows on x86
If we need this function on macOS or Linux, we'll need to shuffle things around a bit.

Reply:
(See Lines 32-34 above)

Reply:
We don't need it on macOS. I think we will need it on Linux. I'd been concentrating on getting it right on Windows because it was failing for other reasons there. I'm looking at Linux now — I suspect it's just a matter of getting the functions inside the correct conditional.